### PR TITLE
chore: Moves Modal to the components folder

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Moves `Modal` to the components folder.

This work is part of [SIP-61](https://github.com/apache/superset/issues/13632)

@rusackas @junlincc

### TEST PLAN
1 - Execute all tests
2 - All tests should pass

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/ImportModal/index.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ import React, { FunctionComponent, useEffect, useRef, useState } from 'react';
 import { styled, t } from '@superset-ui/core';
 
 import Icon from 'src//components/Icon';
-import StyledModal from 'src/common/components/Modal';
+import StyledModal from 'src/components/Modal';
```
Comment (rusackas) [inline superset-frontend/src/components/ImportModal/index.tsx:23]: Not in love with this name (it's inconsistent with the other implementations) which is why I'm not really into default exports, personally. Not a concern for this PR, just whining :)

File: superset-frontend/src/dashboard/components/CrossFilterScopingModal/CrossFilterScopingModal.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@
 import { t } from '@superset-ui/core';
 import React, { FC } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import { StyledModal } from 'src/common/components/Modal';
+import { StyledModal } from 'src/components/Modal';
```
Comment (rusackas) [inline superset-frontend/src/dashboard/components/CrossFilterScopingModal/CrossFilterScopingModal.tsx:22]: This makes my prior comment even more valid, I think... if there's a named `StyledModal` export, we should NOT bring in the default export as `StyledModal`. Anyway...

**Review by rusackas — APPROVED** [review]
LGTM, needs rebase

File: superset-frontend/src/components/ImportModal/index.tsx (hunk 1 above)
Comment (michael-s-molina) [reply superset-frontend/src/components/ImportModal/index.tsx:23]: Don't worry! My next step after organizing the folder will be to create an `index` file and export all components using named exports 😉.

File: superset-frontend/src/dashboard/components/CrossFilterScopingModal/CrossFilterScopingModal.tsx (hunk 2 above)
Comment (michael-s-molina) [reply superset-frontend/src/dashboard/components/CrossFilterScopingModal/CrossFilterScopingModal.tsx:22]: Totally agree. The default export and `StyleModal` have different implementations.

**Review by pkdotson — APPROVED** [review]
LGTG!

**Pull request merged**